packages/esm-amd-loader/test/src/suite.ts

```diff
@@ -256,14 +321,75 @@ suite('meta.url', () => {
 });
 
 // Test for https://github.com/Polymer/tools/issues/335
-suite.skip('dependency ordering', () => {
-  setup(() => {
-    window.executionOrder = []; // only used by these tests.
-  });
+suite('dependency ordering', () => {
   test('all else being equal, dependencies execute in import order', (done) => {
     define(['../race/start.js'], () => {
       assert.deepEqual(window.executionOrder, ['baz', 'foo', 'bar', 'start']);
       done();
     });
   });
+
+  // The dependency graph here looks like:
+  // suite.html#0  -> start-one
+  // suite.html#1  -> start-two
+  // start-one -> [a, e]
+  // start-two -> [a, g, h]
+  // a -> [b, e]
+  // b -> [c, d]
+  // e -> [f, g]
+  // h -> [i, j, k]
+  const testName =
+      'with a deep and more complicated tree of deps, they still execute in import order according to spec';
```

Comment:
what specifically is the test trying to cover with a larger dep tree? Is it just trying to trigger the out-of-order execution due to one chain of deps being longer? Seems like this would be more deterministic and require a small graph if we just setup a server to delay serving of a module earlier in the total order and make sure a module later in the order that doesn't depend on it doesn't run first.

Reply:
Yeah, this test name isn't good. The key thing being tested here but not in the simpler test is that the modules are ordered globally, not locally. i.e. a module isn't ready to run just because it's the first dependency in a define call, it also matters what order its parent runs in.

Updated the test name.